src/app-profile.js

```diff
@@ -35,359 +33,361 @@ var Cluster = require('./cluster.js');
+   *
+   * @example
+   * appProfile.exists(function(err, exists) {});
+   *
+   * //-
+   * // If the callback is omitted, we'll return a Promise.
+   * //-
+   * appProfile.exists().then(function(data) {
+   *   var exists = data[0];
+   * });
+   */
+  exists(gaxOptions, callback) {
+    if (is.fn(gaxOptions)) {
+      callback = gaxOptions;
+      gaxOptions = {};
     }
 
-    callback(null, true);
-  });
-};
+    this.getMetadata(gaxOptions, function(err) {
+      if (err) {
+        if (err.code === 5) {
+          callback(null, false);
+          return;
+        }
 
-/**
- * Get a appProfile if it exists.
- *
- * @param {object} [gaxOptions] Request configuration options, outlined here:
- *     https://googleapis.github.io/gax-nodejs/CallSettings.html.
- *
- * @example
- * appProfile.get(function(err, appProfile, apiResponse) {
- *   // The `appProfile` data has been populated.
- * });
- *
- * //-
- * // If the callback is omitted, we'll return a Promise.
- * //-
- * appProfile.get().then(function(data) {
- *   var appProfile = data[0];
- *   var apiResponse = data[1];
- * });
- */
-AppProfile.prototype.get = function(gaxOptions, callback) {
-  var self = this;
+        callback(err);
+        return;
+      }
 
-  if (is.fn(gaxOptions)) {
-    callback = gaxOptions;
-    gaxOptions = {};
+      callback(null, true);
+    });
   }
 
-  this.getMetadata(gaxOptions, function(err, metadata) {
-    callback(err, err ? null : self, metadata);
-  });
-};
-
-/**
- * Get the app profile metadata.
- *
- * @param {object} [gaxOptions] Request configuration options, outlined
- *     here: https://googleapis.github.io/gax-nodejs/CallSettings.html.
- * @param {function} callback The callback function.
- * @param {?error} callback.err An error returned while making this
- *     request.
- * @param {object} callback.metadata The metadata.
- * @param {object} callback.apiResponse The full API response.
- *
- * @example
- * appProfile.getMetadata(function(err, metadata, apiResponse) {});
- *
- * //-
- * // If the callback is omitted, we'll return a Promise.
- * //-
- * appProfile.getMetadata().then(function(data) {
- *   var metadata = data[0];
- *   var apiResponse = data[1];
- * });
- */
-AppProfile.prototype.getMetadata = function(gaxOptions, callback) {
-  var self = this;
+  /**
+   * Get a appProfile if it exists.
+   *
+   * @param {object} [gaxOptions] Request configuration options, outlined here:
+   *     https://googleapis.github.io/gax-nodejs/CallSettings.html.
+   *
+   * @example
+   * appProfile.get(function(err, appProfile, apiResponse) {
+   *   // The `appProfile` data has been populated.
+   * });
+   *
+   * //-
+   * // If the callback is omitted, we'll return a Promise.
+   * //-
+   * appProfile.get().then(function(data) {
+   *   var appProfile = data[0];
+   *   var apiResponse = data[1];
+   * });
+   */
+  get(gaxOptions, callback) {
+    const self = this;
```

Comment:
Lets file a `type: cleanup` issue to come back and switch to arrow functions and drop the self stuff